### ISSUE TITLE
[Bug]: Tapping into Vue Ecosystem

### ISSUE THREAD

**Issue by MotionPhix** [protonemedia/laravel-splade-core#21]
### What happened?

While in the documentation you are saying *tapping into the Vue ecosystem is easy* by giving the flatpicker example, mine doesn't work when I do the following:

```
<script setup>
import { Menu, MenuButton, MenuItems, MenuItem } from '@headlessui/vue';

const selectedOption = ref(null)

defineOptions({
  availableOptions: {
    type: Array,
    default: () => []
  }
})

function createNewOption() {
const newOption = {
 label: searchQuery,
 value: searchQuery.toLowerCase(),
 };

availableOptions.push(newOption);
 props.selectedOption = newOption;
}

function handleOptionSelection(option) {
   selectedOption.value = option;
}
</script>

<div>
  <div>
    <label for="search">Search options:</label>
    <input v-model="searchQuery" id="search" placeholder="Search options">
  </div>

<Menu>
    <template #default="{ open }">
      <div>
        <MenuButton>
          @{{ selectedOption ? selectedOption.label : 'Select an option' }}
        </MenuButton>

        <MenuItems v-if="open">
          <MenuItem v-for="option in availableOptions" ::key="option.value" @click="handleOptionSelection(option)">
            @{{ option.label }}
          </MenuItem>
        </MenuItems>
      </div>
    </template>
  </Menu>

  <p>Selected Option: @{{ selectedOption ? selectedOption.label : 'None' }}</p>
</div>
```

### How to reproduce the bug

When you add headlessui/vue components, (or any other library that comes with components, this package does not work. fails miserably.

### Package Version

2.2.2

### PHP Version

8.2.0

### Laravel Version

10.0.3

**Comment by bardolf69** [comment]
You did read the docs right... specifically 
***
Why is this experimental?
Although this package comes with an extensive test suite, it's important to note that it's in an experimental stage. It's a new package, and I haven't used it in production yet. Currently, I'm using the [original Splade v1](https://splade.dev/) package for my production needs. However, in 2023 Q4, I plan to integrate this new package into two new projects. It's also worth mentioning that I haven't yet confirmed full compatibility between this package and all the features of the original Splade package.
***

**Comment by pascalbaljet** [comment]
Fixed in v2.4.0
